docs/docs/guides/artifact.mdx

```diff
@@ -45,73 +43,129 @@ nodes:
-- **content type**: the content type for the resulting artifact
-- **targets**: the name of a group of which the members will be a target for this artifact
-- **transformation**: the Jinja2 or Python transformation that should be used
+- **name**: Human-readable description
+- **artifact_name**: Unique identifier (no spaces)
+- **parameters**: Values passed to the transformation query
+- **content_type**: MIME type of the generated artifact
+- **targets**: Group containing target objects
+- **transformation**: Name of the Jinja2 or Python transformation
 
-More details on the `.infrahub.yml` file format can be found in [.infrahub.yml topic](../topics/infrahub-yml.mdx).
+For complete `.infrahub.yml` syntax, see [.infrahub.yml reference](../topics/infrahub-yml.mdx).
 
-Commit the changes to the repository and push them to the Git server
+## Step 4: Deploy the artifact definition
+
+Commit and push your changes to activate the artifact generation:
 
 ```shell
 git add .
 git commit -m "add device_configuration artifact definition"
 git push origin main
 ```
 
-The artifact definition will be created in the database, when the Task worker(s) notice the change in the Git repository. The `device_configuration` should now be visible in the Artifact Definition view in the web interface.
-![Artifact Definition](../media/guides/artifact/artifact_definition.png)
+The task workers will detect the repository change and create the artifact definition in the database.
+
+## Step 5: Verify artifact generation
 
-## Accessing the artifacts
+Check that your artifacts are being generated correctly.
 
-The artifacts are generated by the task-workers. You can find the resulting artifacts in the Artifact view under the **Object Management** menu in the web interface.
+### Through the web interface
 
-![Artifact](../media/guides/artifact/artifact_view.png)
+1. Navigate to **Object Management** → **Artifacts**
+2. Locate your generated artifacts in the list
 
-Open an artifact to get the result.
+![Artifact view](../media/guides/artifact/artifact_view.png)
+
+3. Click on an artifact to view its content
 
 ![Artifact detail](../media/guides/artifact/artifact_detail.png)
-You can download the artifact by clicking on the download button, or by using the REST API endpoint `http://<INFRAHUB_HOST:INFRAHUB_PORT>/api/storage/object/<storage_id>`. The `Storage Id` can be copied from the menu next to the artifact name.
+
+### Through object details
+
+1. Navigate to a specific device (for example, `switch1`)
+2. Select the **Artifacts** tab
+3. View all artifacts generated for this object
+
+## Step 6: Access generated artifacts
+
+Download or retrieve your artifacts using these methods:
+
+### Web interface download
+
+Click the download button on any artifact detail page.
+
+### REST API access
+
+Use the endpoint:
+
+```bash
+http://<INFRAHUB_HOST:INFRAHUB_PORT>/api/storage/object/<storage_id>
```

Comment:
Looks like it was already there before but we shouldn't use the object storage endpoint, to access an artifact, we should use the artifact endpoint instead